simulator/infra/instrcache/instr_cache_memory.h

```diff
@@ -65,6 +65,17 @@ class InstrMemoryCached : public InstrMemory<Instr>
             instr_cache.update( PC, instr);
             return instr;
         }
+
+        size_t memcpy_host_to_guest( Addr dst, const Byte* src, size_t size)
```

Comment:
Not sure why we need that, as they should be inherited

Reply:
Shouln't these interfaces do something with caches? In previous implementation for GDB interfaces, host-to-guest erased cache. And now, I've seen that cache is updated on read (?)

Reply:
Hm, as I understand, cache update is done by cache itself, isn't it?

Reply:
Right, storage of pre-decoded instructions is updated on the read.

Sooner or later, memory will be shared between different devices, and it would be too hard to maintain full coherency of the storage and memory. Instead, I check the coherency on read. If 4 bytes matching the 4 bytes of the storage, I proceed by copying the storage instruction object. If the opposite happened, it means that memory contains a different data and the instruction object is no longer valid.